### PR TITLE
Removed a now unnecessary compatibility fix

### PR DESCRIPTION
**Pull request by danieldouglas92**
A pull request opened during the 2020 hackathon git tutorial. Removed out-dated code regarding different versions of deal.II.

### CODE REVIEW

**Comment by tjhei** [comment]
Your email in your git settings is not registered with github.com (the commit has a grey avatar). Can you add the email to your github account please?

File: source/postprocess/particles.cc [hunk 1]
```diff
@@ -100,18 +100,12 @@ namespace aspect
             // If the property has dim components, we treat it as vector
             if (n_components == dim)
               {
-#if DEAL_II_VERSION_GTE(9,1,0)
+#if DEAL_II_VERSION_GTE(9,2,0)
```
Comment (tjhei) [inline source/postprocess/particles.cc:103]: just remove this line

File: source/postprocess/particles.cc [hunk 2]
```diff
@@ -153,7 +147,7 @@ namespace aspect
         return dataset_names;
       }
 
-#if DEAL_II_VERSION_GTE(9,1,0)
+#if DEAL_II_VERSION_GTE(9,2,0)
```
Comment (tjhei) [inline source/postprocess/particles.cc:150]: and this line

File: source/postprocess/particles.cc [hunk 3]
```diff
@@ -163,16 +157,6 @@ namespace aspect
...
-#else
-      template <int dim>
-      std::vector<std::tuple<unsigned int, unsigned int, std::string> >
-      ParticleOutput<dim>::get_vector_data_ranges () const
-      {
-        return vector_datasets;
-      }
-#endif
-
-    }
```
Comment (tjhei) [inline source/postprocess/particles.cc:175]: This line needs to stay.

File: source/postprocess/particles.cc [hunk 4]
```diff
@@ -100,18 +100,12 @@ namespace aspect
...
                                                           property_index_to_output_index[field_position]+n_components-1,
                                                           field_name,
                                                           DataComponentInterpretation::component_is_part_of_vector));
-#else
-                vector_datasets.push_back(std::make_tuple(property_index_to_output_index[field_position],
-                                                          property_index_to_output_index[field_position]+n_components-1,
-                                                          field_name));
-#endif
               }
-          }
```
Comment (tjhei) [inline source/postprocess/particles.cc:114]: This line needs to stay.

**Comment by bangerth** [comment]
@danieldouglas92 -- ping?

File: source/postprocess/particles.cc (hunk 1 above)
Comment (danieldouglas92) [reply source/postprocess/particles.cc]: Hopefully I've addressed everything!

**Comment by danieldouglas92** [comment]
Thanks for the reminder Wolfgang!

File: source/postprocess/particles.cc [hunk 5]
```diff
@@ -100,18 +100,12 @@ namespace aspect
...
+	  }
```
Comment (bangerth) [inline source/postprocess/particles.cc:108]: Conceptually you did everything right now, except that you've removed one space from before the closing brace here. We have a script that indents everything, and only merge patches that have the exact right indentation. Just add the one space back before the brace here and it should work!

Comment (danieldouglas92) [reply source/postprocess/particles.cc]: Hopefully that should do it!

File: source/postprocess/particles.cc [hunk 6]
```diff
@@ -100,18 +100,12 @@ namespace aspect
...
                                                           field_name,
                                                           DataComponentInterpretation::component_is_part_of_vector));
-#else
-                vector_datasets.push_back(std::make_tuple(property_index_to_output_index[field_position],
-                                                          property_index_to_output_index[field_position]+n_components-1,
-                                                          field_name));
-#endif
               }
-          }
+	   }
```
Comment (bangerth) [inline source/postprocess/particles.cc:108]: There is still a space missing.

To do this automatically, you might want to run `make indent` in your build directory. It indents all source files to the required standard.

**Comment by bangerth** [comment]
No, still not quite yet. Are you using tabs instead of spaces? You can see the current error here: https://github.com/geodynamics/aspect/pull/3662/checks?check_run_id=950217804

**Comment by tjhei** [comment]
Your email in your git settings is still not registered with github.com (the commit has a grey avatar). Can you add the email to your github account please?

**Comment by danieldouglas92** [comment]
I think my email should be linked now! I caught a rogue tab that at the beginning of the line so fingers crossed everything looks good this time around.

**Comment by danieldouglas92** [comment]
Sorry for this! I never would have thought an indent would be such a pain, I'm going to try one more thing I hope there isn't any issues with having multiple commits.

**Comment by tjhei** [comment]
> I think my email should be linked now! I caught a rogue tab that at the beginning of the line so fingers crossed everything looks good this time around.

Half of your commits are good, see https://github.com/geodynamics/aspect/pull/3662/commits

Can you please ask one of us for help to "rebase and squash your commits"?

> I'm going to try one more thing

It looks good now. :+1:

**Comment by bangerth** [comment]
On 8/5/20 1:44 PM, Daniel Douglas wrote:
> Sorry for this! I never would have thought an indent would be such a pain,

Yes, it's a substantial pain to get used to the system. What we found is that 
if indentation is not enforced at the point of merging commits, we end up with 
code that looks like it was written by 100 people with 100 different 
conventions for how to write and indent code (and in some cases where the 
editor doesn't actually enforce anything, with no convention at all). It is 
quite difficult to read such code.

On the plus side, once you know that you have to run `make indent` before 
committing anything, and get used to it, it turns out to be not much of a 
hassle any more.

**Comment by gassmoeller** [comment]
Ah, sorry, one more thing. Could you checkout this branch, and do the following `git commit --amend --reset-author` and then reupload via `git push -f origin remove_dealii_compatibility_fix`? The squash correctly squashed all the commits, but did not overwrite the outdated author information.

**Comment by danieldouglas92** [comment]
Did that do the trick?

**Comment by gassmoeller** [comment]
Perfect! Now lets wait for the testers, if they pass this is ready to merge.

**Comment by tjhei** [comment]
awesome! :+1:

**Pull request merged**